cpp/src/slice2cs/Gen.cpp

```diff
@@ -910,14 +908,29 @@ Slice::Gen::Gen(const string& base, const vector<string>& includePaths, const st
     }
     FileTracker::instance()->addFile(file);
     printHeader();
-
     printGeneratedHeader(_out, fileBase + ".ice");
 
     _out << nl << "#nullable enable";
     _out << sp;
     _out << nl << "[assembly:Ice.Slice(\"" << fileBase << ".ice\")]";
 
     _out << sp;
+
+    /*
```

Comment:
We should comment out printGeneratedHeader above and reenable these pragma from time to time to make sure the generated code is lint-free.

Reply:
What about using DEBUG macro to conditionally enable this. No body is going to use DEBUG compilers in production, we don't ship them but we regularly do debug builds.

Reply:
I don't want to do this for DEBUG builds, x64 builds, macOS builds (...). All these builds should always produce the exact same code.

We could however add a slice2cs compiler flag that:
 - suppress the `<auto-generated>` header
 - generates these pragmas in the .cs code

Something like `--enable-analysis`? As an undocumented option? And we could use it in some CI builds.

Reply:
Adding `--enable-analysis` sounds fine to me.